.devcontainer/Dockerfile

```diff
@@ -98,22 +98,53 @@ RUN git clone https://github.com/libuvc/libuvc.git &&\
     cd ../.. && rm -r libuvc*
 
 
-RUN mkdir -p /opt/lcas/extensions 
-WORKDIR /opt/lcas/extensions 
-RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y 
-RUN git clone https://github.com/eclipse-zenoh/zenoh-plugin-ros2dds.git
-WORKDIR /opt/lcas/extensions/zenoh-plugin-ros2dds
-RUN bash -c "source '$HOME/.cargo/env'; cargo build --release -p zenoh-bridge-ros2dds"
-RUN install target/release/zenoh-bridge-ros2dds /usr/local/bin/
-WORKDIR /
+# RUN mkdir -p /opt/lcas/extensions 
+# WORKDIR /opt/lcas/extensions 
+# RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y 
+# RUN git clone https://github.com/eclipse-zenoh/zenoh-plugin-ros2dds.git
+# WORKDIR /opt/lcas/extensions/zenoh-plugin-ros2dds
+# RUN bash -c "source '$HOME/.cargo/env'; cargo build --release -p zenoh-bridge-ros2dds"
+# RUN install target/release/zenoh-bridge-ros2dds /usr/local/bin/
+# WORKDIR /
+
+# install Zenoh
+RUN mkdir -p /tmp/zenoh-build && \ 
+    cd /tmp/zenoh-build && \
+    (curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y) && \
+    git clone --depth 1 -b 1.0.4 https://github.com/eclipse-zenoh/zenoh-plugin-ros2dds.git && \
+    cd /tmp/zenoh-build/zenoh-plugin-ros2dds && \
+    /bin/bash -c "source '$HOME/.cargo/env'; cargo build --release -p zenoh-bridge-ros2dds" && \
+    install target/release/zenoh-bridge-ros2dds /usr/local/bin/
+
 
 ENV NVIDIA_VISIBLE_DEVICES all
 ENV NVIDIA_DRIVER_CAPABILITIES graphics,utility,compute
 ENV RMW_IMPLEMENTATION=rmw_cyclonedds_cpp
 
-COPY *repos *.sh /tmp/.devcontainer/
-RUN bash /tmp/.devcontainer/install.sh
+COPY .devcontainer/*repos .devcontainer/*.sh /tmp/.devcontainer/
+
+ARG TARGETARCH
+ENV TARGETARCH=${TARGETARCH}
+
+
+COPY . /opt/ros/lcas/src/limo
+RUN apt update && \
+    rosdep init && \
+    rosdep --rosdistro=humble update && \
+    pip install -U argcomplete
+
+RUN cd /opt/ros/lcas && \
+    . /opt/ros/humble/setup.sh && \
+    echo "target: $TARGETARCH" && \
+    if [ ${TARGETARCH} = "arm64" ]; then \
+        # no support for gazebo on arm64
```

Comment:
will need to confirm if this works as expected both on an arm mac and on the actual limo?

Reply:
so, this is a bigger challenge. Gazebo is not supported on jammy on arm64. 

I have made it work though for CMP3103, using a PPA for an arm64 version of plain gazebo and compiling gazebo ROS from source, see https://github.com/LCAS/teaching/blob/2425-devel/.devcontainer/gazebo_ros_pkgs.repos and https://github.com/LCAS/teaching/blob/07e7c6ae4cef6642a26f2b04dc7808bf5d131e5c/.devcontainer/Dockerfile#L14

So, these are changes that would have to be made to any image we use for teaching with simulation on e.g. Apple Mx machines. But all these images to my knowledge include this repository and compile it from source. They don't use the image generated in this repository. The image here is mainly to be deployed on the limos themselves, and there is no need for gazebo. It would just make it more bloated. Hence my decision to not include the steps to compile gazebo into the image generated for arm64. Unless I'm missing something.

Reply:
see https://github.com/gazebosim/gazebo-classic/issues/3236